crates/api/src/site/mod_log.rs

```diff
@@ -42,69 +53,268 @@ impl Perform for GetModlog {
 
     check_private_instance(&local_user_view, context.pool()).await?;
 
+    let type_ = data.type_.unwrap_or(ModlogActionType::All);
     let community_id = data.community_id;
-    let mod_person_id = data.mod_person_id;
+
+    let site = blocking(context.pool(), Site::read_local_site).await??;
+    let (local_person_id, is_admin) = match local_user_view {
+      Some(s) => (s.person.id, is_admin(&s).is_ok()),
+      None => (PersonId(-1), false),
```

Comment:
I spose its fine, but in the future just do a few if lets, its not difficult and you can avoid a whole class of nullpointer bugs. That should always be preferred over making things look compact.